packages/node-cli/src/commands/object/update/UpdateAction.js

```diff
@@ -47,26 +53,90 @@ module.exports = class UpdateAction extends BaseAction {
 			if (params.hasOwnProperty(ANSWERS_NAMES.OVERWRITE_OBJECTS) && !params[ANSWERS_NAMES.OVERWRITE_OBJECTS]) {
 				throw NodeTranslationService.getMessage(MESSAGES.CANCEL_UPDATE);
 			}
-			const sdkParams = CommandUtils.extractCommandOptions(params, this._commandMetadata);
+			let updateCustomRecordsWithInstancesResult = [];
+			if (params.hasOwnProperty(COMMAND_OPTIONS.INCLUDE_INSTANCES) && params[COMMAND_OPTIONS.INCLUDE_INSTANCES]) {
+				const customRecordScriptIds = params[COMMAND_OPTIONS.SCRIPT_ID]
+					.split(' ')
+					.filter((scriptId) => this._isCustomRecordOrSegment(scriptId));
+				params[COMMAND_OPTIONS.SCRIPT_ID] = params[COMMAND_OPTIONS.SCRIPT_ID]
+					.split(' ')
+					.filter((scriptId) => !this._isCustomRecordOrSegment(scriptId))
+					.join(' ');
+				updateCustomRecordsWithInstancesResult = await this._updateCustomRecordWithInstances(params, customRecordScriptIds);
 
-			const executionContextForUpdate = SdkExecutionContext.Builder.forCommand(this._commandMetadata.sdkCommand)
-				.integration()
-				.addParams(sdkParams)
-				.build();
+				delete params[COMMAND_OPTIONS.INCLUDE_INSTANCES];
+			}
 
-			const operationResult = await executeWithSpinner({
-				action: this._sdkExecutor.execute(executionContextForUpdate),
-				message: NodeTranslationService.getMessage(MESSAGES.UPDATING_OBJECTS),
-			});
+			if (params[COMMAND_OPTIONS.SCRIPT_ID] === '') {
+				// If there are no more objects to update (all were custom records) we already return with only previous messages
+				return ActionResult.Builder.withData(updateCustomRecordsWithInstancesResult).build();
+			}
 
-			return operationResult.status === SdkOperationResultUtils.STATUS.SUCCESS
-				? ActionResult.Builder.withData(operationResult.data)
-					.withResultMessage(operationResult.resultMessage)
-					.withCommandParameters(sdkParams)
-					.build()
-				: ActionResult.Builder.withErrors(operationResult.errorMessages).withCommandParameters(sdkParams).build();
+			const sdkParams = CommandUtils.extractCommandOptions(params, this._commandMetadata);
+			const updateObjectsResult = await this._updateObjects(sdkParams);
+			updateCustomRecordsWithInstancesResult = updateCustomRecordsWithInstancesResult.concat(updateObjectsResult.data);
```

Comment:
I personally don't like re-assigning the overall results to the `updateCustomRecordsWithInstancesResult` variable, we could create a new variable named `allResults` or `totalResults` or `aggregatedResults` and assign it to it.
`const aggregatedResults= updateCustomRecordsWithInstancesResult.concat(updateObjectsResult.data)`.

It will also make more clear the Builder call in line 80 that will make use of this variable.

I know it looks I'm going back to the previous implementation, but it's because the context has changed, sorry 😅 .